ext/dynamichmc.jl

```diff
@@ -0,0 +1,86 @@
+###
+### DynamicHMC backend - https://github.com/tpapp/DynamicHMC.jl
+###
+
+import DynamicHMC
+
+# Wraps DynamicHMC as an AbstractSampler
```

Comment:
Is this really needed now that `externalsampler` exists?

Reply:
I think we no longer want to support Gibbs sampling right @yebai?

Reply:
I guess your comment @JaimeRZP was related to me mentioning parameter spaces in https://github.com/TuringLang/Turing.jl/pull/2045#pullrequestreview-1541635867? What about the AD backend choice?